src/models/AppState.js

```diff
@@ -172,7 +172,10 @@ function (Okta, Q, Factor, BrowserFeatures, Errors) {
       // language picker, etc.
       languageCode: ['string', true],
       disableUsername: ['boolean', false, false],
-      trapMfaRequiredResponse: ['boolean', false, false]
+      trapMfaRequiredResponse: ['boolean', false, false],
+      hasFailed: ['boolean', false],
+      errorMessage: ['string', false],
```

Comment:
may we can add all into one object or add proper prefixes.

Reply:
factorVerificationError :{
  errorMessage: ''",
  factor: ['object', false]
}


factorVerificationFailed: 
factorVerificationFailureMessage: